core/shark_turbine/kernel/_support/indexing.py

```diff
@@ -122,20 +115,31 @@ class Grid(metaclass=_GridMeta, symbolic_shape=None):
     """Grid with bounding symbolic shape information in the type."""
 
     symbolic_shape: ClassVar[Optional[SymbolicShapeExpr]]
+    # TODO: dims should also allow dynamic dimensions.
+    dims: list[int]
     rank: int
 
-    def __init__(self, *dims: int):
-        rank = len(dims)
-        if self.symbolic_shape is not None:
-            if rank != len(self.symbolic_shape):
-                raise ValueError(
-                    f"Cannot create {type(self)}({', '.join(str(i) for i in dims)}): mismatched symbolic rank"
-                )
+    def __init__(self):
```

Comment:
Instead of passing the grid bindings as inputs, I am now making the grid find out the symbol bindings from the IndexingContext, and if it cannot find a binding for the symbol, it should consider it a workload binding. Currently, this promotion to a workload binding is not implemented and every symbol is assumed to have a constant value.